test/mapping.test.ts

```diff
@@ -0,0 +1,734 @@
+			expect(mapping.properties.name.type).toEqual('text')
+			// mapping.properties.contact.properties.email.type.should.eql('text')
+			expect(mapping.properties.contact.properties.email.type).toEqual('text')
+			// mapping.properties.contact.properties.tags.type.should.eql('text')
+			expect(mapping.properties.contact.properties.tags.type).toEqual('text')
+			// mapping.properties.contact.properties.should.not.have.property('telephone')
+			expect(mapping.properties.contact.properties).not.toHaveProperty('telephone')
+			// mapping.properties.contact.properties.should.not.have.property('keys')
+			expect(mapping.properties.contact.properties).not.toHaveProperty('keys')
+			done()
+		})
+
+		it('recognizes an multi_field and maps it as one', function (done) {
+			const schema = new Schema({
+				test: {
+					type: String,
+					es_include_in_all: false,
+					es_type: 'multi_field',
+					es_fields: {
+						test: {
+							type: 'text',
+							index: 'analyzed'
+						},
+						untouched: {
+							type: 'text',
+							index: 'not_analyzed'
+						}
+					}
+				}
+			})
+
+			const mapping = generator.generateMapping(schema)
+
+			// mapping.properties.test.type.should.eql('multi_field')
+			expect(mapping.properties.test.type).toEqual('multi_field')
+			// mapping.properties.test.fields.test.type.should.eql('text')
+			expect(mapping.properties.test.fields.test.type).toEqual('text')
+			// mapping.properties.test.fields.test.index.should.eql('analyzed')
+			expect(mapping.properties.test.fields.test.index).toEqual('analyzed')
+			// mapping.properties.test.fields.untouched.type.should.eql('text')
+			expect(mapping.properties.test.fields.untouched.type).toEqual('text')
+			// mapping.properties.test.fields.untouched.index.should.eql('not_analyzed')
+			expect(mapping.properties.test.fields.untouched.index).toEqual('not_analyzed')
+			done()
+		})
+
+		it('recognizes an geo_point and maps it as one', function (done) {
+			const schema = new Schema({
+				geo: {
+					type: String,
+					es_type: 'geo_point'
+				}
+			})
+
+			const mapping = generator.generateMapping(schema)
+
+			// mapping.properties.geo.type.should.eql('geo_point')
+			expect(mapping.properties.geo.type).toEqual('geo_point')
+			done()
+		})
+
+		it('recognizes an geo_point with independent lat lon fields and maps it as one', function (done) {
+			const schema = new Schema({
+				geo_with_lat_lon: {
+					geo_point: {
+						type: String,
+						es_type: 'geo_point',
+						es_lat_lon: true
+					},
+					lat: {
+						type: Number
+					},
+					lon: {
+						type: Number
+					}
+				}
+			})
+
+			const mapping = generator.generateMapping(schema)
+
+			// mapping.properties.geo_with_lat_lon.type.should.eql('geo_point')
+			expect(mapping.properties.geo_with_lat_lon.type).toEqual('geo_point')
+			// mapping.properties.geo_with_lat_lon.lat_lon.should.eql(true)
+			expect(mapping.properties.geo_with_lat_lon.lat_lon).toEqual(true)
```

Comment:
I understand this comments where very useful for the TS/jest migration, but could you remove them before merging?

Reply:
Yes i will remove them.

Reply:
Done.